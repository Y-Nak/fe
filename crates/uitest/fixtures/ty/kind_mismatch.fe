pub struct Foo<T, U> {
    t: T
    u: U
}

pub struct Bar {
    foo: Foo<i32, Foo>
}