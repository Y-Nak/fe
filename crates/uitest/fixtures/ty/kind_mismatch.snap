---
source: crates/uitest/tests/ty.rs
expression: diags
input_file: crates/uitest/fixtures/ty/kind_mismatch.fe
---
error[3-0002]: kind mismatch in type application
  ┌─ kind_mismatch.fe:7:19
  │
7 │     foo: Foo<i32, Foo>
  │                   ^^^ can't apply `(* -> (* -> (* -> (* -> *))))` kind to `(* -> (* -> (* -> *)))` kind


